gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -248,8 +252,15 @@ def run(argv=None):
   variants |= 'FilterVariants' >> filter_variants.FilterVariants(
       reference_names=known_args.reference_names)
   if variant_merger:
-    variants |= (
-        'MergeVariants' >> merge_variants.MergeVariants(variant_merger))
+    if known_args.optimize_for_large_inputs:
+      partitions = variants | 'PartitionVariants' >> beam.Partition(
+          partition_variants.PartitionVariants(), _DEFAULT_PARTITION_NO)
+      merged = [(partitions[i] | 'MergeVariants'+str(i) >>
+                 merge_variants.MergeVariants(variant_merger))
+                for i in xrange(_DEFAULT_PARTITION_NO)]
+      variants = merged | 'FlattenPartitions' >> beam.Flatten()
```

Comment:
Note this implementation does not make any guarantee that each chromosome will be assigned to a separate partition. If reference names do not follow the standard format ('^(c|ch|chr)?([0-9][0-9]?)$') we hash them and randomly assign them to a bucket (without checking for collisions). Even when the reference names have standard format we assign chromosome 1-22 to partitions [0, 21] and the remaining ones (chrX, chrY, chrM, or any variants with unknown_... reference_name) are assigned randomly to other partition indices > 21.

For the use case you mentioned, we will be using the partition_config.yaml file (which is being implemented right now). We will offer a default config file so that users can partition their output table by just setting a flag to the default config file.

Reply:
Thanks Saman for the clarifications. What I am trying to get is to avoid parsing a config file for chromosome based partitioning. Have you thought about generating list of chromosome strings (possibly with count) and feeding it back to the pipeline using side-inputs? I have added both topics related to partitioning to the weekly meeting agenda, we can discuss further there.

Reply:
Bashir, what is the drawback of using a config file? We'd need to implement that logic anyway (and it's not that complicated), but does provide a robust way of partitioning. Note that even the 'hg19 reference' genome has a log of random chromosome names: http://hgdownload.cse.ucsc.edu/goldenpath/hg19/chromosomes/